### PR TITLE
compositor thread: adding corner case check before wait in Draw.

### PR DESCRIPTION
**Pull request by YuanjunHuang**
Adding the check to avoid compositor thread not responsing in some
corner case and leads to UIWDT that found in android stability test.

Jira: https://jira01.devtools.intel.com/browse/OAM-61985
Test: stability test

Signed-off-by: Yuanjun Huang <yuanjun.huang@intel.com>

### CODE REVIEW

**Comment by YuanjunHuang** [comment]
@kalyankondapally

File: common/compositor/compositorthread.cpp [hunk 1]
```diff
@@ -104,6 +104,11 @@ bool CompositorThread::Draw(std::vector<DrawState> &states,
   draw_succeeded_ = true;
   tasks_lock_.unlock();
   Resume();
+
+  if (states_.empty() && media_states_.empty()) {
```
Comment (harishkrupo) [inline common/compositor/compositorthread.cpp]: if both the states are empty, will we ever do any drawing?
If not we could use this as a fast path and return true?

Comment (kalyankondapally) [reply common/compositor/compositorthread.cpp]: in this case we shouldn't be doing any drawing

Comment (harishkrupo) [reply common/compositor/compositorthread.cpp]: So we can check this case as soon as we enter the method and return true?

Comment (kalyankondapally) [reply common/compositor/compositorthread.cpp]: so far, their is no data to show this is needed. Holing on to this PR.

**Comment by kalyankondapally** [comment]
seems to cause some rendering issues, need more testing.
